src/main/java/org/mastodon/mamut/clustering/util/Classification.java

```diff
@@ -1,34 +1,68 @@
 package org.mastodon.mamut.clustering.util;
 
 import com.apporiented.algorithm.clustering.Cluster;
+import org.apache.commons.lang3.tuple.Pair;
 
 import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.stream.Collectors;
 
+/**
+ * A class that encapsulates the result of a clustering algorithm.<p>
+ *     It contains:
+ *     <ul>
+ *         <li>an ordered list of classified objects</li>
+ *         <li>the result of the clustering algorithm</li>
+ *         <li>a mapping from leaf names to objects</li>
+ *         <li>the cutoff value of classification, i.e. where the dendrogram is cut</li>
+ *         <li>the colors of the clusters and the class ids</li>
+ *     </ul>
+ * @author Stefan Hahmann
```

Comment:
I restructured the class `Classification` in accordance to your suggestion.

That allowed:

- getting rid of the obscure "leaf name" variable
- getting rid of any cluster ids

I renamed `algorithmResult` to `rootCluster`, since this is indeed more appropriate.

Cf.: https://github.com/mastodon-sc/mastodon-deep-lineage/commit/3dead27ece04456c751d8526428dec20b13ec1db